### PR TITLE
Update thumbnails while importing a folder

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
While importing a folder dt takes care of the new images but avoids to spam the system with
exposure events and recalculations of the collection.

**But** as reported in #7111 this means there is no visualising of import progress except the progress bar
and this might be seen as annoying.

Unfortunately the whole thing is deeply bound to the way the lighttable has been made so performant.
We don't have a signal that just updates the collection and does a partly recalculation so i found no other
way than to do a brute-force updating.

I found absolutely no hint that database locking - as i suspected first - is related to this problem.

I am really not sure whether this is the "best" we can do right now ... pinging you both

@AlicVB and @TurboGit -- anything better or just don't care?

### CODE REVIEW

File: src/control/jobs/film_jobs.c [hunk 1]
```diff
@@ -253,12 +254,15 @@ static void dt_film_import1(dt_job_t *job, dt_film_t *film)
     g_free(cdn);
 
     /* import image */
-    dt_image_import(cfr->id, (const gchar *)image->data, FALSE);
+    const int imgid = dt_image_import(cfr->id, (const gchar *)image->data, FALSE);
```
Comment (TurboGit) [inline src/control/jobs/film_jobs.c:257]: int32_t

File: src/control/jobs/film_jobs.c [hunk 2]
```diff
@@ -253,12 +254,15 @@ static void dt_film_import1(dt_job_t *job, dt_film_t *film)
...
 
     fraction += 1.0 / total;
     dt_control_job_set_progress(job, fraction);
 
-
+    if((imgid & 7) == 7)
```
Comment (TurboGit) [inline src/control/jobs/film_jobs.c:262]: I would even go with 3. The CPU will mostly be spent in updating the thumbs preview and having 7 to do at the same time but less frequently wont change the fact that thumb preview computing is heavy.

Fact is that recent work from @AlicVB to have a way to use jpeg preview only for a certain size of thumbs will be a good solution here.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some comments, and you also need to do the same for camera/card import. Here is what I have done and is working fine:

```
diff --git a/src/control/jobs/camera_jobs.c b/src/control/jobs/camera_jobs.c
index 50a286dc9..f2e5a5fb5 100644
--- a/src/control/jobs/camera_jobs.c
+++ b/src/control/jobs/camera_jobs.c
@@ -17,6 +17,7 @@
 */
 #include "control/jobs/camera_jobs.h"
 #include "common/darktable.h"
+#include "common/collection.h"
 #include "common/import_session.h"
 #include "common/utility.h"
 #include "control/conf.h"
@@ -282,7 +283,8 @@ void _camera_import_image_downloaded(const dt_camera_t *camera, const char *file
 {
   // Import downloaded image to import filmroll
   dt_camera_import_t *t = (dt_camera_import_t *)data;
-  dt_image_import(dt_import_session_film_id(t->shared.session), filename, FALSE);
+  const int32_t imgid =
+    dt_image_import(dt_import_session_film_id(t->shared.session), filename, FALSE);
   dt_control_queue_redraw_center();
   gchar *basename = g_path_get_basename(filename);
   dt_control_log(ngettext("%d/%d imported to %s", "%d/%d imported to %s", t->import_count + 1),
@@ -293,6 +295,11 @@ void _camera_import_image_downloaded(const dt_camera_t *camera, const char *file
 
   dt_control_job_set_progress(t->job, t->fraction);
 
+ if((imgid & 3) == 3)
+ {
+   dt_collection_update_query(darktable.collection, DT_COLLECTION_CHANGE_NEW_QUERY, NULL);
+ }
+
   if(t->import_count + 1 == g_list_length(t->images))
   {
     // only redraw at the end, to not spam the cpu with exposure events
 
```

File: src/control/jobs/film_jobs.c (hunk 1 above)
Comment (jenshannoschwalm) [reply src/control/jobs/film_jobs.c:257]: Oh man, absolutely right ,,,

**Comment by jenshannoschwalm** [comment]
I didn't do this for camera jobs just because i was not sure this would be the right fix.

**Comment by TurboGit** [comment]
> I didn't do this for camera jobs just because i was not sure this would be the right fix.

Tested on my side, working fine you can go ahead or I can do this after merging this, as you prefer.

**Comment by jenshannoschwalm** [comment]
As you like...

**Comment by TurboGit** [comment]
> As you like...

Go ahead then :)

**Comment by jenshannoschwalm** [comment]
@TurboGit, should be ready to be merged (master and 3.4.1 i guess.)

**Comment by TurboGit** [comment]
Thanks, merged in master and for 3.4.1.

**Comment by jenshannoschwalm** [comment]
I was not sure whether this would be concidered a proper fix for #7111 so i did not use automatic closing of that issue.

**Pull request merged**